### PR TITLE
front: add shortslip distance checkbox in timestops

### LINKED ISSUE #8808: Front: add "short slip distance" check-box next to closed-signal one
Part of https://github.com/osrd-project/osrd-confidential/issues/305#issuecomment-2314801287

In Operational Studies, in the column "Reception on Closed Signal" (réception sur signal fermé):
* Move current checkbox on the left
* If checked, add a little checkbox on the right labelled (2 lines?) "short slip dist." ("faible gliss."), with the hover "Reception on closed signal with short slip distance" ("Arrêt sur signal fermé à faible glissement").
* Route the result to editoast using the enum replacing the closed-signal flag from https://github.com/OpenRailAssociation/osrd/issues/8809

### PR DESCRIPTION
**Pull request by RomainValls**
closes #8808

### CODE REVIEW

File: front/src/modules/timesStops/helpers/utils.ts [hunk 1]
```diff
@@ -303,26 +305,28 @@ export function calculateStepTimeAndDays(
...
+    return shortSlipDistance ? 'SHORT_SLIP_STOP' : 'STOP';
   }
   return 'OPEN';
 }
 
 /** Convert receptionSignal enum to onStopSignal boolean */
-export function receptionSignalToOnStopSignal(
-  receptionSignal: ReceptionSignal | undefined
-): boolean | undefined {
+export function receptionSignalToSignalBooleans(receptionSignal: ReceptionSignal | undefined) {
```
Comment (SharglutDev) [inline front/src/modules/timesStops/helpers/utils.ts]:
```suggestion
export function receptionSignalToSignalBooleans(receptionSignal?: ReceptionSignal) {
```

File: front/src/modules/timesStops/helpers/utils.ts [hunk 2]
```diff
@@ -303,26 +305,28 @@ export function calculateStepTimeAndDays(
 
 /** Convert onStopSignal boolean to receptionSignal enum */
 export function onStopSignalToReceptionSignal(
-  onStopSignal: boolean | undefined
+  onStopSignal: boolean | undefined,
+  shortSlipDistance: boolean | undefined
```
Comment (SharglutDev) [inline front/src/modules/timesStops/helpers/utils.ts]:
```suggestion
  onStopSignal?: boolean,
  shortSlipDistance?: boolean
```

File: front/src/modules/timesStops/hooks/useTimeStopsColumns.ts [hunk 3]
```diff
@@ -116,6 +116,14 @@ export const useTimeStopsColumns = <T extends TimeStopsRow>(
         disabled: ({ rowData, rowIndex }) =>
           isOutputTable || (rowIndex !== allWaypoints.length - 1 && !rowData.stopFor),
       },
+      {
+        ...keyColumn('shortSlipDistance', checkboxColumn as Partial<Column<boolean | undefined>>),
+        title: t('shortSlipDistance'),
+        ...fixedWidth(94),
+        disabled: ({ rowData, rowIndex }) =>
+          isOutputTable ||
+          (rowIndex !== allWaypoints.length - 1 && !rowData.stopFor && !rowData.onStopSignal),
```
Comment (SharglutDev) [inline front/src/modules/timesStops/hooks/useTimeStopsColumns.ts]: Should be enough as `onStopSignal` is already disabled if no `stopFor`

```suggestion
          (rowIndex !== allWaypoints.length - 1 && !rowData.onStopSignal),
```

File: front/public/locales/fr/timesStops.json
Comment (SharglutDev) [inline front/public/locales/fr/timesStops.json:1]: In the issue, there is something about a message displayed on hover `Arrêt sur signal fermé à faible glissement`, is it still wanted or not ? Did you check with @thibautsailly ?

**Review by SharglutDev — COMMENTED** [review]
Thank you for the PR, left some comments

**Review by Akctarus — APPROVED** [review]
tested, works well

File: front/public/locales/fr/timesStops.json
Comment (RomainValls) [reply front/public/locales/fr/timesStops.json:1]: It doesn't seem possible to add a message displayed on hover with the react datasheet grid library

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested. The name of the columns aren't exactly like in the issue but I guess it's on purpose

**Pull request merged**